File: SupplySense_Backend/backend/app/schemas/purchase_order.py
"""
SupplySense — Purchase Order Pydantic v2 Schemas
=================================================
"""

from typing import Optional, List
from decimal import Decimal
from datetime import date
from pydantic import BaseModel, Field, ConfigDict


class PurchaseOrderItemSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str = Field(..., description="Item ID.")
    product_id: str = Field(..., description="Product ID.")
    product_name: Optional[str] = Field(default=None, description="Product name.")
    sku: Optional[str] = Field(default=None, description="Product SKU.")
    quantity: int = Field(..., description="Ordered quantity.")
    unit_price: Decimal = Field(..., description="Unit purchase price.")
    total_price: Decimal = Field(..., description="Line total price.")


class PurchaseOrderResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str = Field(..., description="PO ID.")
    supplier_id: str = Field(..., description="Supplier ID.")
    supplier_name: Optional[str] = Field(default=None, description="Supplier company name.")
    warehouse_id: str = Field(..., description="Target Warehouse ID.")
    warehouse_name: Optional[str] = Field(default=None, description="Target Warehouse name.")
    order_date: date = Field(..., description="PO issuance date.")
    expected_delivery_date: Optional[date] = Field(default=None, description="Expected SLA delivery date.")
    status: str = Field(..., description="Draft, Pending, Approved, Shipped, Delivered, Rejected.")
    priority: Optional[str] = Field(default="Normal", description="Normal, High, Urgent.")
    approved_by: Optional[str] = Field(default=None, description="Approving manager username.")
    total_amount: Decimal = Field(..., description="Total PO value.")


class PurchaseOrderDetailResponse(PurchaseOrderResponse):
    items: List[PurchaseOrderItemSchema] = Field(default_factory=list, description="PO line items.")

File: SupplySense_Frontend/src/types/purchase-order.ts
/**
 * SupplySense — Purchase Order Type Definitions
 * Maps to backend: backend/app/schemas/purchase_order.py
 */

export interface PurchaseOrderItem {
  id: string;
  product_id: string;
  product_name?: string;
  sku?: string;
  quantity: number;
  unit_price: number;
  total_price: number;
}

export interface PurchaseOrder {
  id: string;
  supplier_id: string;
  supplier_name?: string;
  warehouse_id: string;
  warehouse_name?: string;
  order_date: string;
  expected_delivery_date?: string;
  status: string;
  priority?: "Normal" | "High" | "Urgent";
  approved_by?: string;
  total_amount: number;
  items?: PurchaseOrderItem[];
}

export interface CreatePOItemInput {
  product_id: string;
  quantity: number;
  unit_price?: number;
}

export interface CreatePOInput {
  supplier_id: string;
  warehouse_id: string;
  expected_delivery_date?: string;
  priority?: "Normal" | "High" | "Urgent";
  notes?: string;
  items: CreatePOItemInput[];
}
